docs/presentations/pycon2024/maintainers.qmd

```diff
@@ -0,0 +1,322 @@
+  - Phillip Cloud
+execute:
+  echo: true
+format:
+  revealjs:
+    footer: <https://ibis-project.org>
+    chalkboard: true
+    # https://quarto.org/docs/presentations/revealjs/themes.html#using-themes
+    theme: dark
+---
+
+# what
+
+## maybe this is you
+
+![](./docker-eye-roll.gif){fig-align="center"}
+
+## or this
+
+![](./wonka.png){fig-align="center"}
+
+## or maybe even this
+
+![](./basement-ci.jpeg){fig-align="center"}
+
+## this talk isn't earth shattering
+
+opinions informed by experience
+
+:::: {.columns}
+
+::: {.column width="50%"}
+### overview
+
+- what we learned about maintenance building Ibis
+- day to day of supporting 20+ databases
+- unique challenges
+:::
+
+::: {.column width="50%"}
+### topics
+
+- some docker stuff
+- some packaging stuff
+- pytest plugins you should be using
+:::
+::::
+
+# quick overview of Ibis
+
+## Ibis is a Python library for:
+
+- exploratory data analysis (EDA)
+- analytics
+- data engineering
+- ML preprocessing
+- building your own dataframe library
+
+::: {.r-fit-text}
+dev to prod with the same API
+:::
+
+## Getting started...
+
+```{python}
+#| code-fold: true
+#| echo: false
+
+import ibis
+
+t = ibis.examples.penguins.fetch()
+t.to_parquet("penguins.parquet")
+```
+
+```{python}
+import ibis
+
+ibis.options.interactive = True
+```
+
+```{python}
+t = ibis.read_parquet("penguins.parquet")
+t.head(3)
+```
+
+## One API, 20+ backends {.smaller .scrollable}
+
+::: {.panel-tabset}
+
+## DuckDB
+
+```{python}
+con = ibis.connect("duckdb://")
+```
+
+```{python}
+t = con.read_parquet("penguins.parquet")
+t.head(3)
+```
+
+```{python}
+t.group_by("species", "island").agg(count=t.count()).order_by("count")
+```
+
+## Polars
+
+```{python}
+con = ibis.connect("polars://")
+```
+
+```{python}
+t = con.read_parquet("penguins.parquet")
+t.head(3)
+```
+
+```{python}
+t.group_by("species", "island").agg(count=t.count()).order_by("count")
+```
+
+## DataFusion
+
+```{python}
+con = ibis.connect("datafusion://")
+```
+
+```{python}
+t = con.read_parquet("penguins.parquet")
+t.head(3)
+```
+
+```{python}
+t.group_by("species", "island").agg(count=t.count()).order_by("count")
+```
+
+## PySpark
+
+```{python}
+con = ibis.connect("pyspark://")
+```
+
+```{python}
+t = con.read_parquet("penguins.parquet")
+t.head(3)
+```
+
+```{python}
+t.group_by("species", "island").agg(count=t.count()).order_by("count")
+```
+
+## 16+ other things
+
+![](./machine.gif){fig-align="center" width="100%" height="100%"}
+
+:::
+
+## how it works
+
+Ibis compiles down to SQL or dataframe code:
+
+```{python}
+#| echo: false
+
+import os
+import sys
+sys.path.append(os.path.abspath(".."))
+
+from backends_sankey import fig
+fig.show()
+```
+
+# what's in Ibis?
+
+## by the numbers {.smaller}
+
+:::: {.columns}
+::: {.column width="50%"}
+### backends
+- **17** SQL
+- **3** non-SQL
+- **2** cloud
+:::
+
+::: {.column width="50%"}
+### engines + APIs
+- **9** distributed SQL
+- **3** dataframe
+- oldest: **~45** years 👀
```

Comment:
are you gonna say which this is? makes me curious...I guess postgres?

Reply:
Oracle